core/src/main/java/org/elasticsearch/cluster/DiskUsage.java

```diff
@@ -44,6 +51,40 @@ public DiskUsage(String nodeId, String nodeName, String path, long totalBytes, l
         this.path = path;
     }
 
+    public DiskUsage(StreamInput in) throws IOException {
+        this.nodeId = in.readString();
+        this.nodeName = in.readString();
+        this.path = in.readString();
+        this.totalBytes = in.readVLong();
+        this.freeBytes = in.readVLong();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(this.nodeId);
+        out.writeString(this.nodeName);
+        out.writeString(this.path);
+        out.writeVLong(this.totalBytes);
+        out.writeVLong(this.freeBytes);
+    }
+
+    public XContentBuilder toShortXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field("path", this.path);
+        builder.byteSizeField("total_bytes", "total", this.totalBytes);
+        builder.byteSizeField("used_bytes", "used", this.getUsedBytes());
+        builder.byteSizeField("free_bytes", "free", this.freeBytes);
+        builder.field("free_disk_percent", this.getFreeDiskAsPercentage());
+        builder.field("used_disk_percent", this.getUsedDiskAsPercentage());
```

Comment:
> No matter what precision you pick will have that problem, for example rounding 89.99 to one decimal point will round to 90, and so on.

Sure, but it's a bigger difference in terms of space, if we rounded to integer a 1% difference is ~10gb (assuming a 1tb disk), which is still quite a bit of space, if we rounded to one decimal it becomes ~1gb, or ~100mb for two decimals. 10gb per percentage is quite a coarse grain